### PR TITLE
Return the skipped tokens (tree-sitter) information to semgrep-cli

### PR DESCRIPTION
**Pull request by aryx**
test plan:
make test

```
$ semgrep-core -l ts -e '$X == $X' skip_tokens.ts  -json
+ /home/pad/yy/_build/default/src/cli/Main.exe -l ts -e '$X == $X' skip_tokens.ts -json
.
{"matches":[],"errors":[{"rule_id":"-e/-f","error_type":["PartialParsing",[{"path":"skip_tokens.ts","start":{"line":2,"col":15,"offset":0},"end":{"line":2,"col":16,"offset":1}},{"path":"skip_tokens.ts","start":{"line":9,"col":15,"offset":0},"end":{"line":9,"col":16,"offset":1}}]],"severity":"warning","location":{"path":"skip_tokens.ts","start":{"line":2,"col":15,"offset":0},"end":{"line":2,"col":16,"offset":1}},"message":"`)` was unexpected"}],"skipped":[],"stats":{"okfiles":0,"errorfiles":1}}
```
PR checklist:

- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by aryx** [comment]
This will also help https://github.com/returntocorp/semgrep/issues/2186

File: semgrep-core/src/core/Xtarget.ml [hunk 1]
```diff
@@ -27,5 +27,5 @@ type t = {
   lazy_content : string lazy_t;
   (* This is valid only for xlang = Xlang.L ..., not for LRegex|LGeneric *)
   lazy_ast_and_errors :
-    (AST_generic.program * Semgrep_error_code.error list) lazy_t;
+    (AST_generic.program * Parse_info.token_location list) lazy_t;
```
Comment (IagoAbal) [inline semgrep-core/src/core/Xtarget.ml:30]: Don't we want to allow for any kind of error here? We could always filter the ones that are interesting for the CLI later. I am trying to update DeepSemgrep and we have:

```
    try
    let { Parse_target.ast; skipped_tokens; _} = 
          Parse_target.parse_and_resolve_name
          lang file in
    file, (ast, skipped_tokens)
    with
    | exn -> file, ([], [exn_to_error file exn])
```
What to do with that `exn` then?

Comment (aryx) [reply semgrep-core/src/core/Xtarget.ml:30]: ok, I'll try to update deep-semgrep and see if we need to go revert some code in semgrep.

Comment (aryx) [reply semgrep-core/src/core/Xtarget.ml:30]: Here the only errors we were getting were always skipped_tokens, so I preferred to be precise.
The parsing function can also return exns (e.g., ParseError, LexicalError, etc.), but once we get an AST,
the only possible thing attached to it are skipped_tokens.

**Pull request merged**